### PR TITLE
use forked redis and check external TALKER_AGENT_VERSION structure

### PR DESCRIPTION
**Pull request by assafgi**

### CODE REVIEW

File: talker_agent/install_agent.sh [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
 
 if [ "$DOWNLOAD_TALKER" = true ] ; then
+  # talker agent version can be:
+  #       tag            - for example: 1.8.2
+  #       tag-commit_id  - for example: 1.8.2-02ee1aed8275c79a5d0840505caaaef0fa53b26e
+  if [[ $TALKER_AGENT_VERSION == *"-"* ]]; then
+    IFS='-' read -r -a parts <<< "$TALKER_AGENT_VERSION"
+    commit_id=${parts[1]}
+  else
+    commit_id=v$TALKER_AGENT_VERSION
```
Comment (DoronCohen) [inline talker_agent/install_agent.sh]: no point in calling it `commit-id` if it's not a commit

File: talker_agent/install_agent.sh [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
+  fi
+
   sudo mkdir -p /root/talker
-  cd /root/talker
-  curl -sfL https://github.com/weka-io/talker/raw/v1.8.1/agent/talker-service > talker-service
-  curl -sfL https://github.com/weka-io/talker/raw/v1.8.1/agent/talker.py > talker.py
+  cd /root/talker || exit
```
[the remark this answers is not in the document]
Comment (assafgi) [reply talker_agent/install_agent.sh]: [link_from_pycharm](https://github.com/koalaman/shellcheck/wiki/SC2164)

File: talker_agent/install_agent.sh (hunk 1 above)
Comment (assafgi) [reply talker_agent/install_agent.sh]: use `version` ?

File: talker_agent/install_agent.sh [hunk 3]
```diff
@@ -1,17 +1,27 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (DoronCohen) [inline talker_agent/install_agent.sh]: use `set -e`. this will fail the script if any of the commands fail. including `cd ...`

**Pull request merged**